docs/http-cloud-provider.md

```diff
@@ -0,0 +1,643 @@
+|resources:quantity:format|The format in which you want the amount to be stored. The types of formats are restricted to the list below.|"DecimalSI"|
+
+Resource Types
+
+|Name|Description|
+|----|-----------|
+|cpu|The CPU amount in cores.|
+|memory|The amount of memory allocated, in bytes.|
+|storage|The Volume size, in bytes.|
+|pods|The number of pods.|
+|services|The number of services.|
+|replicationcontrollers|The number of Replication Controllers.|
+|resourequotas|The number of resource quotas.|
+|secrets|The number of resource secrets.|
+|persistentvolumeclaims|The number of resource persistent volume claims.|
+
+Format Types
+
+|Name|Description|Example|
+|----|-----------|-------|
+|DecimalExponent|The number is in exponential form.|12e6|
+|BinarySI|The number is in binary SI form.|12Mi (12 * 2^20)|
+|DecimalSI|The number is stored in Dcimal SI form.|12M (12 * 10^6)|
+Example
+
+```json
+{
+    "resources": [
+        {
+            "resourceName": "cpu",
+            "quantity":
+                {
+                    "amount": 4,
+                    "format": "DecimalSI"
+                }
+        },
+        {
+            "resourceName": "memory",
+            "quantity":
+                {
+                    "amount": 1048576,
+                    "format": "DecimalSI"
+                }
+        }
+    ]
+}
+```
+
+#### GET /v1/instances/node/{hostName}
+
+Returns the name of the node which is related to the specific host.
+
+##### Expected return value:
+
+|Name|Description|Example Value|
+|----|-----------|-------------|
+|nodeName|The name of the node in the host|"my-node"|
+Example
+
+```json
+{
+    "nodeName": "my-node"
+}
+```
+
+#### POST /v1/instances/sshKey
+
+Adds the SSH Key provided to all the instances. Returns whether the key was added or not.
+
+##### Sender body:
+
+|Name|Description|Example Value|
+|----|-----------|-------------|
+|user|The user's whose SSH key needs to be added.|"name"|
+|keydata|The actual ssh key. Any format allowed not restricted to strings.|"zPjoihsswRTGIUHKLHIHO345@435"|
+Example
+
+```json
+{
+    "user": "name",
+    "keyData": "zPjoihsswRTGIUHKLHIHO345@435"
+}
+```
+
+##### Expected Return Value:
+
+|Name|Description|Example Value|
+|----|-----------|-------------|
+|SSHKeyAdded|Boolean value regarding whether the SSH Key was added or not.|false|
+Example
+
+```json
+{
+    "SSHKeyAdded": false
+}
+```
+
+## TCPLoadBalancer
+
+The TCPLoadBalancer interface allows the kubernetes platform to query the cloud provider about the Load Balancer the cloud provider has and requests to change the load balancer as required.
+
+### Methods
+
+#### GET /v1/tcpLoadBalancer/supported
+
+Know whether the interface is supported or not.
+
+##### Expected return value:
+
+|Name|Description|Example Value|
+|----|-----------|-------------|
+|supported|Whether the interface is supported or not.|true|
+Example
+
+```json
+{
+    "supported":false
+}
+```
+
+#### GET /v1/tcpLoadBalancer/{region}/{name}
```

Comment:
Why region? Why not availability zone? Why is tcpLoadBalancer namespaced but instances is not?